core/src/main/java/fr/sncf/osrd/api/PathfindingResult.java

```diff
@@ -85,35 +85,36 @@ public static List<PathWaypointResult> getWaypointsOnRoute(
             if (offset + range.getLength() >= routeRange.start()) {
                 // Truncate the range to match the part of the route we use
                 var truncatedRange = truncateTrackRange(range, offset, routeRange);
-
-                // List all waypoints on the track range in a tree map, with offsets from the range start as key
-                var waypoints = new ArrayList<PathWaypointResult>();
-
-                // Add operational points as optional waypoints
-                for (var op : truncatedRange.getOperationalPoints())
-                    waypoints.add(PathWaypointResult.suggestion(op.element(), trackSection, op.offset()));
-
-                // Add user defined waypoints
-                var truncatedRangeOffset = offset + Math.abs(truncatedRange.getStart() - range.getStart());
-                for (int i = 0; i < waypointsOffsetsList.size(); i++) {
-                    var trackRangeOffset = waypointsOffsetsList.get(i) - truncatedRangeOffset;
-
-                    // We can have tiny differences here because we accumulate offsets in a different way
-                    if (Math.abs(trackRangeOffset - truncatedRange.getLength()) < POSITION_EPSILON)
-                        trackRangeOffset = truncatedRange.getLength();
-                    if (Math.abs(trackRangeOffset) < POSITION_EPSILON)
-                        trackRangeOffset = 0;
-
-                    if (trackRangeOffset >= 0 && trackRangeOffset <= truncatedRange.getLength()) {
-                        var location = truncatedRange.offsetLocation(trackRangeOffset);
-                        waypoints.add(PathWaypointResult.userDefined(location, trackRangeOffset));
-                        waypointsOffsetsList.remove(i--); // Avoids duplicates on track transitions
+                if (truncatedRange != null) {
```

Comment:
can you use == null and continue instead?

Reply:
The loop ends with `offset += range.getLength();` and we don't want to skip that